data/transactions/logic/README.md

```diff
@@ -0,0 +1,211 @@
+| `sha512_256` | SHA512_256 hash of value, yields [32]byte |
+| `ed25519verify` | for (data, signature, pubkey) verify the signature of the data against the pubkey => {0 or 1} |
+| `rand` | push random uint64 to stack. based on block.Seed and PCG RNG |
+| `+` | A plus B. Panic on overflow. |
+| `-` | A minus B. Panic if B > A. |
+| `/` | A divided by B. Panic if B == 0. |
+| `*` | A times B. Panic on overflow. |
+| `<` | A less than B => {0 or 1} |
+| `>` | A greater than B => {0 or 1} |
+| `<=` | A less than or equal to B => {0 or 1} |
+| `>=` | A greater than or equal to B => {0 or 1} |
+| `&&` | A is not zero and B is not zero => {0 or 1} |
+| `\|\|` | A is not zero or B is not zero => {0 or 1} |
+| `==` | A is equal to B => {0 or 1} |
+| `!=` | A is not equal to B => {0 or 1} |
+| `!` | X == 0 yields 1; else 0 |
+| `len` | yields length of byte value |
+| `btoi` | converts bytes as big endian to uint64 |
+| `%` | A modulo B. Panic if B == 0. |
+| `\|` | A bitwise-or B |
+| `&` | A bitwise-and B |
+| `^` | A bitwise-xor B |
+| `~` | bitwise invert value |
+
+### Loading Values
+
+Opcodes for getting data onto the stack.
+
+Some of these have immediate data in the byte or bytes after the opcode.
+
+| Op | Description |
+| --- | --- |
+| `intcblock` | load block of uint64 constants |
+| `intc` | push value from uint64 constants to stack by index into constants |
+| `intc_0` | push uint64 constant 0 to stack |
+| `intc_1` | push uint64 constant 1 to stack |
+| `intc_2` | push uint64 constant 2 to stack |
+| `intc_3` | push uint64 constant 3 to stack |
+| `bytecblock` | load block of byte-array constants |
+| `bytec` | push bytes constant to stack by index into constants |
+| `bytec_0` | push bytes constant 0 to stack |
+| `bytec_1` | push bytes constant 1 to stack |
+| `bytec_2` | push bytes constant 2 to stack |
+| `bytec_3` | push bytes constant 3 to stack |
+| `arg` | push LogicSig.Args[N] value to stack by index |
+| `arg_0` | push LogicSig.Args[0] to stack |
+| `arg_1` | push LogicSig.Args[1] to stack |
+| `arg_2` | push LogicSig.Args[2] to stack |
+| `arg_3` | push LogicSig.Args[3] to stack |
+| `txn` | push field from current transaction to stack |
+| `gtxn` | push field to the stack from a transaction in the current transaction group |
+| `global` | push value from globals to stack |
+| `load` | copy a value from scratch space to the stack |
+| `store` | pop a value from the stack and store to scratch space |
+
+**Transaction Fields**
+
+| Index | Name | Type |
+| --- | --- | --- |
+| 0 | Sender | []byte |
+| 1 | Fee | uint64 |
+| 2 | FirstValid | uint64 |
+| 3 | LastValid | uint64 |
+| 4 | Note | []byte |
+| 5 | Receiver | []byte |
+| 6 | Amount | uint64 |
+| 7 | CloseRemainderTo | []byte |
+| 8 | VotePK | []byte |
+| 9 | SelectionPK | []byte |
+| 10 | VoteFirst | uint64 |
+| 11 | VoteLast | uint64 |
+| 12 | VoteKeyDilution | uint64 |
+| 13 | Type | []byte |
+| 14 | TypeEnum | uint64 |
```

Comment:
Where are the TypeEnum values enumerated?

Reply:
under the op docs for `txn`, noted that